sim/dex-formats.ts

```diff
@@ -813,6 +813,31 @@ export class DexFormats {
 
 		ruleTable.resolveNumbers(format, this.dex);
 
+		const canMegaEvo = this.dex.gen <= 7 || ruleTable.has('+pokemontag:past');
+		if (ruleTable.has('obtainableformes') && canMegaEvo) {
+			// Banning Rayquaza-Mega implicitly adds Mega Rayquaza Clause
+			const mray = this.dex.species.get('rayquazamega');
+			let mrayBanned = false;
+			const banReason = ruleTable.check('pokemon:' + mray.id);
+			if (banReason) {
+				mrayBanned = true;
+			} else if (banReason !== '') {
+				for (const ruleid of ruleTable.tagRules) {
+					if (ruleid.startsWith('*')) continue;
+					const tag = Tags[ruleid.slice(12)];
+					if (tag.speciesFilter?.(mray)) {
+						if (ruleid.startsWith('+')) break;
+						mrayBanned = true;
+						break;
+					}
+				}
+			}
+			if (mrayBanned) {
```

Comment:
Unless I'm missing something...
```suggestion
			if (ruleTable.isBannedSpecies(mray)) {
```
Although at this point it might make more sense for the Mega Rayquaza Clause to ban Rayquaza-Mega and have `canMegaEvo` check whether it's banned.

Reply:
isBannedSpecies checks the base forme as well, which isn't desirable here. However, I suppose `isBannedSpecies('rayquazamega') && !isBannedSpecies('rayquaza')` would work, and better, too.